keps/sig-node/0035-20190130-topology-manager.md

```diff
@@ -315,18 +315,26 @@ index efbd72c133..f86a1a5512 100644
  }
 
 +message TopologyInfo {
-+  optional int32 socketID = 1 [default = -1];
++  NUMANode node = 1;
++}
++
++message NUMANode {
++    int64 ID = 1;
 +}
 +
  /* E.g:
  * struct Device {
  *    ID: "GPU-fef8089b-4820-abfc-e83e-94318197576e",
+ *    State: "Healthy",
++ *    Topology: 
++ *      Node: 
++ *        ID: 1 
 @@ -85,6 +89,8 @@ message Device {
  	string ID = 1;
  	// Health of the device, can be healthy or unhealthy, see constants.go
  	string health = 2;
-+	// Topology details of the device (optional.)
-+	optional TopologyInfo topology = 3;
++	// Topology details of the device
++	TopologyInfo topology = 3;
```

Comment:
so, topology info is not anymore optional for the plugin?

Reply:
In Proto3 every field is optional, so the y keyword is not required.